guides/user/pipeline/expressions/index.md

```diff
@@ -10,492 +10,356 @@ redirect_from:
-
-![](12.png)
+evaluates to _Hello-randomString-world_.
 
-When your stage executes, you will see a dropdown with the provided choice. 
+## How do I use a pipeline expression?
 
-![](13.png)
+### Where can I use pipeline expressions?
 
-In subsequent stages, you can reference the choice provided via the [#judgment( stage Name ) ](#judgmentstring) helper method `${ #judgment( 'manual judgment stage name' )}`. 
+You can use pipeline expressions in any place in the UI where you can enter
+free-form text, with the exception of the pipeline configuration stage. If you
+want to set the value of a field using a pipeline expression but there is no
+text box available, you can use the [Edit as
+JSON](/guides/user/pipeline/managing-pipelines/#edit-a-pipeline-as-JSON)
+pipeline feature.
 
-### External sources
+### When are pipeline expressions evaluated?
 
-You can also read from external sources via the [#fromUrl( url )](#fromurlstring) and [#jsonFromUrl( url )](#jsonfromurlstring) helper methods. 
+Spinnaker evaluates the expressions for a given stage at the beginning of that
+stage. You can't use pipeline expressions during the pipeline configuration
+stage, because Spinnaker doesn't begin evaluating expressions until after the
+pipeline configuration stage has finished.
 
-## Pipeline expression language reference
+## What tools do I have for writing pipeline expressions?
 
-The Pipeline Expression syntax is implemented with the Spring Expression Language ( SpEL ). You can learn more about SPeL by following this [link](http://docs.spring.io/spring/docs/current/spring-framework-reference/html/expressions.html).
+### Helper functions
 
-### Writing expressions
+There are several built-in [helper functions](/reference/pipeline/expressions/#helper-functions)
+to simplify some common use cases. These functions allow you to do things like
+access a particular stage by name, strip non-alphanumerical characters out of a
+string, or parse JSON.
 
-A pipeline expression is started by `${` and terminated by `}` : 
+Adding a pound sign (`#`) within your pipeline expression displays a list of all
+the helper functions that are available.
 
-`${ expression here }`
+![](images/helper-func-autocomplete.png)
 
-You can interpolate expressions and strings, so
+### Helper properties
 
-`${expressionA}-randomString-${expressionB}` 
+_Helper properties_ are variables which refer to global information about the
+current pipeline execution. For example, `execution` is a variable which refers
+to the current pipeline execution, whereas `trigger` lets you access information
+about the pipeline trigger. For example, if you've set up a Jenkins trigger and
+want to know which build triggered the current pipeline, you could access that
+information with the expression `${trigger["buildInfo"]["number"]}`. For a
+complete list of available properties, see the reference guide.
```

Comment:
Then you can probably delete this last sentence because you've linked to the reference already?